pkg/kubelet/server/server.go

```diff
@@ -378,20 +378,17 @@ func (s *Server) InstallDefaultHandlers(enableCAdvisorJSONEndpoints bool) {
 		compbasemetrics.HandlerFor(r, compbasemetrics.HandlerOpts{ErrorHandling: compbasemetrics.ContinueOnError}),
 	)
 
-	// deprecated endpoint which will be removed in release 1.20.0+.
-	s.addMetricsBucketMatcher("metrics/resource/v1alpha1")
-	v1alpha1ResourceRegistry := compbasemetrics.NewKubeRegistry()
-	v1alpha1ResourceRegistry.CustomMustRegister(stats.NewPrometheusResourceMetricCollector(s.resourceAnalyzer, stats.Config()))
-	s.restfulCont.Handle(path.Join(resourceMetricsPath, v1alpha1.Version),
-		compbasemetrics.HandlerFor(v1alpha1ResourceRegistry, compbasemetrics.HandlerOpts{ErrorHandling: compbasemetrics.ContinueOnError}),
-	)
-
 	s.addMetricsBucketMatcher("metrics/resource")
 	resourceRegistry := compbasemetrics.NewKubeRegistry()
 	resourceRegistry.CustomMustRegister(collectors.NewResourceMetricsCollector(s.resourceAnalyzer))
 	s.restfulCont.Handle(resourceMetricsPath,
 		compbasemetrics.HandlerFor(resourceRegistry, compbasemetrics.HandlerOpts{ErrorHandling: compbasemetrics.ContinueOnError}),
 	)
+	// deprecated endpoint which will be removed in release 1.20.0+. Returns same as above metrics/resource API.
+	s.addMetricsBucketMatcher("metrics/resource/v1alpha1")
+	s.restfulCont.Handle(path.Join(resourceMetricsPath, v1alpha1.Version),
+		compbasemetrics.HandlerFor(resourceRegistry, compbasemetrics.HandlerOpts{ErrorHandling: compbasemetrics.ContinueOnError}),
+	)
```

Comment:
I think this is what we want to do? Hook up the alpha endpoint to the stable endpoint.

Alpha endpoint had to move below the existing one to delete the registry and pick up the stable registry endpoint.

Reply:
> I think this is what we want to do?

No, what we want to do is DO NOT touch the original endpoint `metrics/resource/v1alpha1` and add a new endpoint to replace the original one.